topics/reference_guide/custom_language_support/websymbols_static_symbols.md

```diff
@@ -0,0 +1,424 @@
+
+Directly under `contributions` property are listed namespaces with their contributions.
+Currently only `html`, `css` or `js` namespaces are allowed, however in the future this limitation
+will be lifted to support Web Types for other technologies.
+
+The `namespace` object contains symbol kind names listed as properties.
+Some symbol kinds are predefined and directly supported by IDE (see Symbol Kinds/Supported for reference).
+The kind of a symbol should relate to its role.
+For instance a Vue directive should be of the kind `vue-directives`.
+Each framework should define a set of custom symbol kinds if needed.
+Reference for the most important symbol kinds defined by frameworks supported by IDEs is below.
+
+Each symbol kind name property of a namespace object is an array of symbol contributions.
+A symbol contribution should have at least a name.
+Contributions in addition to standard properties can define sub-contributions and custom properties.
+A custom property is a JSON property, whose value is of a simple type (string, number or boolean),
+or is an array of simple types.
+If a contribution’s JSON property’s value is an object value or an array of objects, it is treated as a list of sub-contributions.
+Such contributions will be assigned to the same namespace as the parent contributions.
+To use a different namespace for sub-contributions, nest symbol kind JSON property name under a `js`, `css` or `html` property. E.g:
+
+```JSON
+{
+  "contributions": {
+    "html": {
+      "elements": [
+        {
+          "name": "my-element",
+          "description": "A custom HTML element",
+          "attributes": [
+            {
+              "name": "foo",
+              "description": "A custom HTML attribute of `my-element`"
+            }
+          ],
+          "css": {
+            "properties": [
+              {
+                "name": "--bg-color",
+                "description": "Background color of my-element"
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+In the example below, Web Types contributes information that the `my-element` HTML element supports a custom CSS property `--bg-color`.
+The `attributes` are implicitly under the `html` namespace.
+To contribute `foo` attribute one could also write it in longer form:
+
+```JSON
+{
+  "name": "my-element",
+  "description": "A custom HTML element",
+  "html": {
+    "attributes": [
+      {
+        "name": "foo",
+        "description": "A custom attribute of `my-element`"
+      }
+    ]
+  }
+}
+```
+
+Each Web Types contribution is represented in the Web Symbols framework by a
+[`PsiSourcedWebSymbol`](%gh-ic%/platform/webSymbols/src/com/intellij/webSymbols/PsiSourcedWebSymbol.kt)
+object.
+All of the Web Types contributions are mapped 1-1 and custom properties are accessible through `properties` property.
+
+## Including Web Types
+
+Web Types can currently be discovered by the IDE in following ways:
+
+{style="narrow"}
```

Comment:
Unfortunately, the narrow style doesn't fit many of the property/method names.

Please remove it, and let's use the standard one. It's less concise but is less evil, IMHO.

Reply:
I agree.

@YannCebron Do you also agree?

Are we changing **all** definition lists to keep the page consistent or only the ones that look ugly?

Reply:
we must use same style across whole page for consistency

it seems `{style="medium"}` would work quite well except for `getModificationCount()`...

Reply:
Alright, here is the compromise: We use "full" now and when we refactor the method descriptions into API doc, we can probably change everything to "narrow" or "medium".